pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -267,23 +269,84 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		reqLogger.Info("Created Service", "Service", serDNS)
 
 		if infinispan.IsExposed() {
-			externalService := r.serviceExternal(infinispan)
-			if externalService != nil {
-				err = r.client.Create(context.TODO(), externalService)
-				if err != nil && !errors.IsAlreadyExists(err) {
-					reqLogger.Error(err, "failed to create external Service", "Service", ser)
+			switch infinispan.Spec.Expose.Type {
+			case infinispanv1.ExposeTypeLoadBalancer, infinispanv1.ExposeTypeNodePort:
+				externalService := r.serviceExternal(infinispan)
+				if externalService != nil {
+					err = r.client.Create(context.TODO(), externalService)
+					if err != nil && !errors.IsAlreadyExists(err) {
+						reqLogger.Error(err, "failed to create external Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					err = r.client.Update(context.TODO(), infinispan)
+					if err != nil {
+						reqLogger.Error(err, "Failed to update Infinispan with Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					reqLogger.Info("Created External Service", "Service", externalService)
+				} else {
+					reqLogger.Info("External Service NOT created. Unsupported type?", "Service", nil)
+
+				}
+			case infinispanv1.ExposeTypeRoute:
+				lsService := ispncom.LabelsResource(infinispan.ObjectMeta.Name, "infinispan-service-external")
+				route := openshiftv1.Route{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Route",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      infinispan.Name,
+						Namespace: infinispan.Namespace,
+						Labels:    lsService,
+					},
+					Spec: openshiftv1.RouteSpec{To: openshiftv1.RouteTargetReference{Kind: "Service",
+						Name: infinispan.Name},
```

Comment:
@rigazilla do we need to define port for Route?

Reply:
yeah, since we already have port and host field I think it's ok to configure them also for route if provided by the user

Reply:
It looks like service port, not custom defined and actual if multiple service ports are available. In out case it's just single but may be multiple in the future.

Reply:
well, actually: `host` is ok. `port` is probably beyond a basic configuration for route/ingressa